components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImplTest.java

```diff
@@ -632,48 +644,84 @@ public void testCreateAndGetApplicationWithProtocolService() throws IdentityAppl
-            if (ApplicationConstants.StandardInboundProtocols.OAUTH2.equals(inboundAuthenticationRequestConfig
-                    .getInboundAuthType())) {
-                Assert.assertEquals(inboundAuthenticationRequestConfig.getInboundAuthKey(),
-                        "auth key");
-            }
-            // This is the newly created inbound protocol configuration. Validate the newly created inbound protocol
-            // is added.
-            if (ApplicationConstants.StandardInboundProtocols.SAML2.equals(inboundAuthenticationRequestConfig
-                    .getInboundAuthType())) {
-                Assert.assertEquals(inboundAuthenticationRequestConfig.getInboundAuthKey(),
-                        APPLICATION_INBOUND_AUTH_KEY_1);
+        if (!isOrganization) {
+            /*
+             There should be 2 inbound protocol configurations. The one that already exists and the one that
+             is created.
+            */
+            Assert.assertEquals(applicationByResourceId.getInboundAuthenticationConfig()
+                    .getInboundAuthenticationRequestConfigs().length, 2);
+            for (InboundAuthenticationRequestConfig inboundAuthenticationRequestConfig : applicationByResourceId
+                    .getInboundAuthenticationConfig().getInboundAuthenticationRequestConfigs()) {
+                /*
+                 This is the existing inbound protocol configuration. Validate the existing inbound protocol
+                 configuration is unchanged.
+                */
+                if (ApplicationConstants.StandardInboundProtocols.OAUTH2.equals(inboundAuthenticationRequestConfig
+                        .getInboundAuthType())) {
+                    Assert.assertEquals(inboundAuthenticationRequestConfig.getInboundAuthKey(),
+                            APPLICATION_INBOUND_AUTH_KEY_OAUTH2);
+                }
+                /*
+                 This is the newly created inbound protocol configuration. Validate the newly created inbound
+                 protocol is added.
+                */
+                if (ApplicationConstants.StandardInboundProtocols.SAML2.equals(inboundAuthenticationRequestConfig
+                        .getInboundAuthType())) {
+                    Assert.assertEquals(inboundAuthenticationRequestConfig.getInboundAuthKey(),
+                            APPLICATION_INBOUND_AUTH_KEY_1);
+                }
             }
+            applicationManagementService.deleteApplications(SUPER_TENANT_ID);
+        } else {
+            /*
+             There should be 1 inbound protocol configurations since in the sub organization level, saml inbound
+             protocol configurations are not allowed.
+            */
+            Assert.assertEquals(applicationByResourceId.getInboundAuthenticationConfig()
+                    .getInboundAuthenticationRequestConfigs().length, 1);
+            applicationManagementService.deleteApplications(SUPER_TENANT_ID);
```

Comment:
shouldn't this be moved after line number 714?

Reply:
Also since it is common to if / else both, move it outside

Reply:
moved